source/remoteClient/dialogs.py

```diff
@@ -0,0 +1,336 @@
+		evt.Skip()
+		self.panel.Destroy()
+		if self.clientOrServer.GetSelection() == 0:
+			self.panel = ClientPanel(parent=self.container)
+		else:
+			self.panel = ServerPanel(parent=self.container)
+		self.mainSizer.Fit(self)
+
+	def onOk(self, evt: wx.CommandEvent) -> None:
+		if self.clientOrServer.GetSelection() == 0 and (
+			not self.panel.host.GetValue() or not self.panel.key.GetValue()
+		):
+			gui.messageBox(
+				# Translators: A message box displayed when the host or key field is empty and the user tries to connect.
+				_("Both host and key must be set."),
+				# Translators: A title of a message box displayed when the host or key field is empty and the user tries to connect.
+				_("Error"),
+				wx.OK | wx.ICON_ERROR,
+			)
+			self.panel.host.SetFocus()
+		elif (
+			self.clientOrServer.GetSelection() == 1
+			and not self.panel.port.GetValue()
+			or not self.panel.key.GetValue()
+		):
+			gui.messageBox(
+				# Translators: A message box displayed when the port or key field is empty and the user tries to connect.
+				_("Both port and key must be set."),
+				# Translators: A title of a message box displayed when the port or key field is empty and the user tries to connect.
+				_("Error"),
+				wx.OK | wx.ICON_ERROR,
+			)
+			self.panel.port.SetFocus()
+		else:
+			evt.Skip()
+
+	def getKey(self) -> str:
+		return self.panel.key.GetValue()
+
+	def getConnectionInfo(self) -> ConnectionInfo:
+		if self.clientOrServer.GetSelection() == 0:  # client
+			host = self.panel.host.GetValue()
+			serverAddr, port = protocol.addressToHostPort(host)
+			mode = ConnectionMode.MASTER if self.connectionType.GetSelection() == 0 else ConnectionMode.SLAVE
+			return ConnectionInfo(
+				hostname=serverAddr,
+				mode=mode,
+				key=self.getKey(),
+				port=port,
+				insecure=False,
+			)
+		else:  # server
+			port = int(self.panel.port.GetValue())
+			mode = "master" if self.connectionType.GetSelection() == 0 else "slave"
+			return ConnectionInfo(
+				hostname="127.0.0.1",
+				mode=mode,
+				key=self.getKey(),
+				port=port,
+				insecure=True,
+			)
+
+
+class CertificateUnauthorizedDialog(wx.MessageDialog):
+	def __init__(self, parent: Optional[wx.Window], fingerprint: Optional[str] = None):
+		# Translators: A title bar of a window presented when an attempt has been made to connect with a server with unauthorized certificate.
+		title = _("NVDA Remote Connection Security Warning")
+		message = _(
+			# Translators: {fingerprint} is a SHA256 fingerprint of the server certificate.
+			"Warning! The certificate of this server could not be verified.\nThis connection may not be secure. It is possible that someone is trying to overhear your communication.\nBefore continuing please make sure that the following server certificate fingerprint is a proper one.\nIf you have any questions, please contact the server administrator.\n\nServer SHA256 fingerprint: {fingerprint}\n\nDo you want to continue connecting?",
```

Comment:
Can we tighten up this message a bit?
```suggestion
			"""The certificate of this server could not be verified. Using the wrong fingerprint may allow a third party to access the remote session.
			
			Before continuing, please make sure that the following server certificate fingerprint is correct.
			Server SHA256 fingerprint: {fingerprint}
			
			Continue connecting anyway?""",
```